### ISSUE TITLE
scale_colour/fill aesthetic only works for British spelling

### ISSUE THREAD

**Issue by jvcasillas** [tidyverse/ggplot2#2674]
The latest addition of `aesthetics` to `scale_fill_*` and `scale_colour_*` are fantastic, but the colour geom only works with the British spelling. 

```
ggplot(iris, aes(x = Sepal.Length, y = Petal.Length, color = Sepal.Length, fill = Petal.Length)) +
  geom_point(shape = 21, size = 3, stroke = 2) +
  scale_color_viridis_c(name = "Length",
                         option = "B", aesthetics = c("color", "fill"))
```

Produces...

<img width="856" alt="screen shot 2018-05-30 at 09 26 06" src="https://user-images.githubusercontent.com/1747068/40722887-8589fbf2-63eb-11e8-924e-f2cafa737997.png">

When the desired output is...

<img width="860" alt="screen shot 2018-05-30 at 09 26 41" src="https://user-images.githubusercontent.com/1747068/40722912-98e490ea-63eb-11e8-8701-03667e265b8b.png">

**Comment by clauswilke** [comment]
Yes, that's a limitation of the current code. The required changes are a bit more involved and won't make it into 2.3.0. I already opened an issue about this: #2649, so I'll close this one.
